src/apply_rules/object_rules.rs

```diff
@@ -0,0 +1,508 @@
+                        .values_mut()
+                        .filter(|ticket| ticket.object_type == ModelObjectType::Network)
+                        .filter(|ticket| &ticket.object_id == grouped)
+                        .for_each(|mut ticket| ticket.object_id = network_id.to_string());
+                    networks_to_remove.insert(grouped.to_string());
+                }
+                network_rule = true;
+            }
+        }
+        if !network_rule {
+            report.add_error(
+                format!("The rule on the \"{}\" network was not applied", network_id),
+                TransitModelReportCategory::ObjectNotFound,
+            );
+        } else if !collections.networks.contains_id(network_id) {
+            new_networks.push(serde_json::from_value(properties.clone())?);
+        }
+    }
+    collections
+        .networks
+        .retain(|cm| !networks_to_remove.contains(&cm.id));
+
+    collections.networks.extend(new_networks);
+
+    Ok(collections)
+}
+
+fn read_networks_rules_file<P: AsRef<Path>>(
+    report: &mut Report<TransitModelReportCategory>,
+    network_rules_file: P,
+) -> Option<Vec<ObjectProperties>> {
+    info!("Reading networks rules");
+
+    #[derive(Debug, Deserialize)]
+    struct NetworksRule {
+        #[serde(rename = "networks")]
+        networks_rules: Vec<ObjectProperties>,
+    }
+
+    match File::open(network_rules_file) {
+        Ok(file) => {
+            let rdr: StdResult<NetworksRule, SerdeJsonError> = serde_json::from_reader(file);
+            match rdr {
+                Ok(val) => Some(val.networks_rules),
+                Err(e) => {
+                    if !(e.classify() == Category::Data) {
+                        report.add_error(format!("{}", e), TransitModelReportCategory::InvalidFile);
+                    }
+                    None
+                }
+            }
+        }
+        Err(_) => None,
+    }
+}
+
+fn apply_rules_on_networks<P: AsRef<Path>>(
+    networks_rules_file: P,
+    lines_by_network: &HashMap<String, IdxSet<Line>>,
+    collections: Collections,
+    mut report: &mut Report<TransitModelReportCategory>,
+) -> Result<Collections> {
+    let networks_rules = read_networks_rules_file(&mut report, networks_rules_file);
+    match networks_rules {
+        Some(res) => {
+            check_and_apply_networks_rules(&mut report, collections, res, lines_by_network)
+        }
+        None => {
+            info!("no rule on network provided");
+            Ok(collections)
+        }
+    }
+}
+
+pub(crate) fn apply_rules<P: AsRef<Path>>(
+    object_rules_file: P,
+    lines_by_network: &HashMap<String, IdxSet<Line>>,
+    lines_by_commercial_mode: &HashMap<String, IdxSet<Line>>,
+    vjs_by_physical_mode: &HashMap<String, IdxSet<VehicleJourney>>,
+    mut collections: Collections,
+    mut report: &mut Report<TransitModelReportCategory>,
+) -> Result<Collections> {
+    collections = apply_rules_on_networks(
```

Comment:
Loading the file once into a Rust structure and send this structure to the other functions.

```rust
let configuration = parse_configuration(object_rules_file);
```